### PR TITLE
feat: Internal User Policy

### PR DESCRIPTION
**Pull request by saffaalvi**
Resolves https://github.com/StatCan/daaas/issues/922

### CODE REVIEW

**Comment by sylus** [comment]
This looks really good @saffaalvi just two minor nits.

- Can we have a new line after markdown heading
- Can the main README.md be updated

@zachomedia can you review as well?

**Comment by blairdrummond** [comment]
@saffaalvi We should make the folder name the same as the policy name, just for clarity

Also one thing I'd like to check: The subject of a rolebinding or auth policy is assumed to be an email address here, but that may not be the case. I.e. this will deny (in some circumstances) ALL role bindings.

Let's add a quick line to the rego to check "is the subject an email address" before creating a violation

File: general/internal-user/template.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+                type: string
+  targets:
+    - target: admission.k8s.gatekeeper.sh
+      rego: |
+        package denyexternalusers
+
+        violation[{"msg": msg}] {
+          obj := input.review.object
+          obj.kind == "RoleBinding"
+          email := obj.subjects[0].name
```
Comment (blairdrummond) [inline general/internal-user/template.yaml:30]: You can replace "0" with "_" to apply over the whole array, which is better

File: general/internal-user/template.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+          not isEmployee(email)
```
Comment (blairdrummond) [inline general/internal-user/template.yaml:31]: As mentioned in the other comment

internal = isEmail && not isEmployee

We just need to add the isEmail logic here

File: general/internal-user/template.yaml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+          profile := data.inventory.cluster["kubeflow.org/v1"]["Profile"][ns]
+          profileLabel := input.parameters.labels[_]
+          profile.metadata.labels[profileLabel] == "true"
+          msg := sprintf("Profile %v has %v=%v", [ns, profileLabel, profile.metadata.labels[profileLabel]])
+        }
+
+        violation[{"msg": msg}] {
+          obj := input.review.object
+          obj.kind == "AuthorizationPolicy"
+          email := obj.spec.rules[0]["when"][0]["values"][0]
```
Comment (blairdrummond) [inline general/internal-user/template.yaml:42]: I would use _ instead of 0 to range over all rules

**Review by blairdrummond — CHANGES_REQUESTED** [review]
Small changes to the array indexes and email checking

**Comment by saffaalvi** [comment]
@sylus I'll be working on this today so will try my best to have them done by EOD!

**Review by blairdrummond — APPROVED** [review]
LGTM!

**Comment by sylus** [comment]
This looks great @saffaalvi :D Awesome work!

**Pull request merged**